### ISSUE TITLE
Streamline the redirect situation

### ISSUE THREAD

**Issue by afranke** [matrix-org/matrix.org#1109]
We have too many different ways to set up redirects (so that legacy URLs still work but land you on a newer page):

* in the Apache config (in another repo managed by ops)
* in a handful of leftover jekyll pages (those won’t be a problem for long)
* in ~50 gatsby pages (`$ git grep 'http-equiv.*refresh' -- "./gatsby/*"`)
* and worst of all we have [a script that generates another 50](https://github.com/matrix-org/matrix.org/blob/master/scripts/generate-redirects.js).

We don’t need to settle on a single one (I reckon some belong in Apache config and some don’t), but we probably want to merge the last three methods.

**Comment by afranke** [comment]
The script seems to be what created most of the gatsby ones actually. Nothing in the repo calls it and it must have been called manually, as a one-off. I’m a bit confused now.

**Comment by richvdh** [comment]
> We have too many different ways to set up redirects (so that legacy URLs still work but land you on a newer page):
> 
>     * in the Apache config (in another repo managed by ops)

Internal users can see these at https://github.com/matrix-org/matrix-ansible-private/blob/main/roles/apache/templates/sites/matrix.org.conf.j2#L78

>     * in a handful of leftover jekyll pages (those won’t be a problem for long)
> 
>     * in ~50 gatsby pages (`$ git grep 'http-equiv.*refresh' -- "./gatsby/*"`)
>     * and worst of all we have [a script that generates another 50](https://github.com/matrix-org/matrix.org/blob/master/scripts/generate-redirects.js).

per the above, these are the same thing. It sounds like the script has served its purpose and can be ignored or removed.

Sounds like the jekyll ones are going away too.

So that leaves us with two places:
 * Apache
 * Gatsby

> I reckon some belong in Apache config and some don’t

yes, this. Each of Apache/Gatsby has its own attraction:

 * Importantly, the Apache redirects are "server-side" (they return a 30x code rather than relying on the HTML engine) - which means they also work for things like the federation tester which serves a lump of json. They are also generally lighter weight.
 * The Gatsby ones allow the redirects to be versioned alongside the content and don't require management from the ops team.

In short, I think there is a place for both. I'm sure there are some that could do with being moved from one to the other, though.

**Comment by richvdh** [comment]
I don't think there's anything to do here.

**Comment by afranke** [comment]
The 54 results found by `git grep 'http-equiv.*refresh' -- "./gatsby/*"` ought to be turned into lines in a `.htaccess` file, providing a central place to track them instead of the current scattered mess that makes it hard to follow.

**Comment by richvdh** [comment]
mm, maybe.

**Comment by thibaultamartin** [comment]
Since the Zola migration, redirects to legacy urls are done in CloudFlare (via https://github.com/matrix-org/matrix.org/blob/main/static/_redirects)

A handful of them still exist as `aliases` in Zola and should probably make it to the `_redirects` file
